src/sentry/static/sentry/app/views/projectSavedSearches.jsx

```diff
@@ -173,33 +143,68 @@ const ProjectSavedSearches = createReactClass({
     });
   },
 
-  handleRemovedSearch(data) {
-    let savedSearchList = this.state.savedSearchList;
-    this.setState({
-      savedSearchList: savedSearchList.filter(search => {
-        return search.id !== data.id;
-      }),
-    });
-  },
+  handleUpdate(params) {
+    let {orgId, projectId} = this.props.params;
+    let loadingIndicator = IndicatorStore.add(t('Saving changes..'));
+    let {data, isDefault, isUserDefault} = params;
+    let key = typeof isDefault !== 'undefined' ? 'isDefault' : 'isUserDefault';
+    let {savedSearchList} = this.state;
+    let newSearchList = savedSearchList.map(search => ({
+      ...search,
+      [key]: data.id === search.id,
+    }));
 
-  handleDefaultSearch(data) {
-    let savedSearchList = this.state.savedSearchList;
-    savedSearchList.forEach(search => {
-      search.isDefault = data.id === search.id;
-    });
-    this.setState({
-      savedSearchList,
-    });
+    this.setState(
+      {
+        savedSearchList: newSearchList,
+      },
+      () => {
+        this.api.request(`/projects/${orgId}/${projectId}/searches/${data.id}/`, {
+          method: 'PUT',
+          data: {
+            isDefault,
+            isUserDefault,
+          },
+          error: () => {
+            this.setState({
```

Comment:
does error or anything called for cancelled requests? (like lets say it's slow and they navigate away)

sure it's fine here just curious about the way we treat this (optimistic updates) in general considering how api mixin behaves

Reply:
Yeah, aborted requests do trigger error handler